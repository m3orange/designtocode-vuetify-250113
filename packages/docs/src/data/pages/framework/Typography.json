{
  "title": "header",
  "titleText": "headerText",
  "mdLink": "https://material.io/design/typography/the-type-system.html",
  "children": [
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "helperHeader"
        },
        {
          "type": "text",
          "lang": "helperText"
        },
        {
          "type": "api",
          "value": [
            "v-display-4",
            "v-display-3",
            "v-display-2",
            "v-display-1",
            "v-headline",
            "v-title",
            "v-subtitle-2",
            "v-subtitle-1",
            "v-body-2",
            "v-body-1",
            "v-caption",
            "v-overline"
          ]
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "fontHeader"
        },
        {
          "type": "text",
          "lang": "fontText"
        },
        {
          "type": "markdown",
          "lang": "fonts"
        },
        {
          "type": "style-headings"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "weightsHeader"
        },
        {
          "type": "text",
          "lang": "weightsText"
        },
        {
          "type": "markdown",
          "lang": "weights"
        },
        {
          "type": "text",
          "lang": "italicText"
        },
        {
          "type": "markdown",
          "lang": "italic"
        },
        {
          "type": "style-weights"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "transformHeader"
        },
        {
          "type": "text",
          "lang": "transformText"
        },
        {
          "type": "markdown",
          "lang": "transforms"
        },
        {
          "type": "style-typefaces"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "wrapHeader"
        },
        {
          "type": "text",
          "lang": "wrapText"
        },
        {
          "type": "markdown",
          "lang": "wraps"
        },
        {
          "type": "example",
          "value": "truncate"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "up-next",
          "value": [
            "framework/alignment",
            "framework/content"
          ]
        }
      ]
    }
  ]
}
