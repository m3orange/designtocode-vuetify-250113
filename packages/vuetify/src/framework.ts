import { createDisplay, VuetifyDisplaySymbol } from './composables/display'
import { createTheme, VuetifyThemeSymbol } from './composables/theme'
import { defaultSets, VuetifyIconSymbol } from '@/composables/icons'
import { createDefaults, VuetifyDefaultsSymbol } from '@/composables/defaults'
import { createLocaleAdapter, VuetifyLocaleAdapterSymbol } from '@/composables/locale'
import { createRtl, VuetifyRtlSymbol } from '@/composables/rtl'
import { aliases, mdi } from '@/iconsets/mdi'

// Utilities
import { mergeDeep } from '@/util'

// Types
import type { App, ComponentPublicInstance, InjectionKey } from 'vue'
import type { DisplayOptions } from '@/composables/display'
import type { ThemeOptions } from '@/composables/theme'
import type { IconOptions } from '@/composables/icons'
import type { LocaleAdapter, LocaleOptions } from '@/composables/locale'
import type { RtlOptions } from '@/composables/rtl'
import type { DefaultsOptions } from '@/composables/defaults'
import { reactive } from 'vue'

export interface VuetifyInstance {}

export interface VuetifyOptions {
  components?: Record<string, any>
  directives?: Record<string, any>
  defaults?: DefaultsOptions
  display?: DisplayOptions
  theme?: ThemeOptions
  icons?: IconOptions
  locale?: (LocaleOptions & RtlOptions) | (LocaleAdapter & RtlOptions)
}

export const VuetifySymbol: InjectionKey<VuetifyInstance> = Symbol.for('vuetify')

export const useVuetify = () => {
  const vuetify = inject(VuetifySymbol)

  if (!vuetify) {
    throw new Error('Vuetify has not been installed on this app')
  }

  return vuetify
}

export const createVuetify = (options: VuetifyOptions = {}) => {
  const install = (app: App) => {
    const {
      components = {},
      directives = {},
      icons = {},
    } = options

    for (const key in directives) {
      const directive = directives[key]

      app.directive(key, directive)
    }

    for (const key in components) {
      const component = components[key]

      app.component(key, component)
    }

    const vuetify = {}

    app.provide(VuetifySymbol, vuetify)
    app.provide(VuetifyDefaultsSymbol, createDefaults(options.defaults))
    app.provide(VuetifyDisplaySymbol, createDisplay(options.display))
    app.provide(VuetifyThemeSymbol, createTheme(options.theme))
    app.provide(VuetifyIconSymbol, mergeDeep({
      defaultSet: 'mdi',
      sets: {
        ...defaultSets,
        mdi,
      },
      aliases,
    }, icons))
    const { adapter, rootInstance } = createLocaleAdapter(app, options?.locale)
    app.provide(VuetifyLocaleAdapterSymbol, adapter)
    app.provide(VuetifyRtlSymbol, createRtl(rootInstance, options?.locale))

    function inject (this: ComponentPublicInstance, key: InjectionKey<any> | string) {
      const vm = this.$

      const provides = vm.parent?.provides ?? vm.vnode.appContext?.provides

      if (provides && (key as any) in provides) {
        return provides[(key as string)]
      }
    }

    app.mixin({
      computed: {
        $vuetify () {
          return reactive({
            defaults: inject.call(this, VuetifyDefaultsSymbol),
            display: inject.call(this, VuetifyDisplaySymbol),
            theme: inject.call(this, VuetifyThemeSymbol),
            icons: inject.call(this, VuetifyIconSymbol),
            locale: inject.call(this, VuetifyLocaleAdapterSymbol),
            rtl: inject.call(this, VuetifyRtlSymbol),
          })
        },
      },
    })
  }

  return { install }
}
