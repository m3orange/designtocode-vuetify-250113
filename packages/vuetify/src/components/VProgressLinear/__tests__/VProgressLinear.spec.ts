// Components
import { VProgressLinear } from '../'

// Utilities
import { mount } from '@vue/test-utils'
import { createVuetify } from '@/framework'
import { h, toRef } from '@vue/runtime-core'

describe('VProgressLinear', () => {
  const vuetify = createVuetify()
  const mountFunction = (options?: any) => mount(VProgressLinear, {
    global: {
      plugins: [vuetify],
    },
    ...options,
  })

  it('should render component and match snapshot', async () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 33,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ modelValue: -1, bufferValue: -1 })
    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ modelValue: 101, bufferValue: 101 })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render inactive component and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 33,
        active: false,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component in RTL mode', () => {
    const wrapper = mount(VProgressLinear, {
      props: {
        modelValue: 33,
      },
      global: {
        plugins: [createVuetify({ locale: { defaultLocale: 'ar' } })],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render reversed component', () => {
    const wrapper = mountFunction({
      props: {
        reverse: true,
        modelValue: 33,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render reverse component in RTL mode', () => {
    const wrapper = mount(VProgressLinear, {
      props: {
        modelValue: 33,
        reverse: true,
      },
      global: {
        plugins: [createVuetify({ locale: { defaultLocale: 'ar' } })],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 33,
        color: 'orange',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with css color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 33,
        color: '#336699',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with color and background opacity and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 33,
        color: 'orange',
        bgOpacity: 0.5,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with color and background color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 33,
        color: 'orange',
        bgColor: 'blue',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with color and background color and opacity and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 33,
        color: 'orange',
        bgColor: 'blue',
        bgOpacity: 0.5,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render indeterminate progress and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        indeterminate: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render indeterminate progress with query prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        indeterminate: true,
        query: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with buffer value and match snapshot', async () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 33,
        bufferValue: 80,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({
      bufferValue: 0,
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with buffer value and value > buffer value and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 90,
        bufferValue: 80,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render default slot content', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: 90,
        bufferValue: 80,
      },
      slots: {
        default: () => h('div', { class: 'foobar' }, 'content'),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should respond to click events', async () => {
    const wrapper = mountFunction({
      attachToDocument: true,
      props: {
        modelValue: 0,
      },
    })

    const rect = wrapper.vm.$el.getBoundingClientRect()

    wrapper.vm.$el.getBoundingClientRect = () => ({
      ...rect,
      width: 1000,
    })

    await wrapper.trigger('click', { offsetX: 200 })

    expect(wrapper.emitted('update:modelValue')).toEqual([
      [20],
    ])
  })

  it('should render a stream component', () => {
    const wrapper = mountFunction({
      props: {
        stream: true,
      },
    })

    expect(wrapper.find('.v-progress-linear__stream')).toBeTruthy()
  })
})
