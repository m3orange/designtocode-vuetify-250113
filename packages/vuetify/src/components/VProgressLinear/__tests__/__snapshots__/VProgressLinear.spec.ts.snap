// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VProgressLinear should render component and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; left: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component and match snapshot 2`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="0"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; left: 0%; width: 0%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 0%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component and match snapshot 3`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="100"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; left: 100%; width: 0%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 100%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component in RTL mode 1`] = `
<div class="v-progress-linear v-progress-linear--reverse v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; right: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component with buffer value and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; left: 33%; width: 47%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component with buffer value and match snapshot 2`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background primary"
       style="opacity: 0.3; left: 33%; width: 0%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component with buffer value and value > buffer value and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="90"
>
  <div class="v-progress-linear__background primary"
       style="opacity: 0.3; left: 90%; width: 0%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 90%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component with color and background color and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-blue"
       style="opacity: 1; left: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-orange"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component with color and background color and opacity and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-blue"
       style="opacity: 0.5; left: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-orange"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component with color and background opacity and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-orange"
       style="opacity: 0.5; left: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-orange"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component with color and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-orange"
       style="opacity: 0.3; left: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-orange"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render component with css color and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background"
       style="background-color: rgb(51, 102, 153); opacity: 0.3; left: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate"
         style="background-color: rgb(51, 102, 153); width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render default slot content 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="90"
>
  <div class="v-progress-linear__background primary"
       style="opacity: 0.3; left: 90%; width: 0%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 90%;"
    >
    </div>
  </transition-stub>
  <div class="v-progress-linear__content">
    <div class="foobar">
      content
    </div>
  </div>
</div>
`;

exports[`VProgressLinear should render inactive component and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 0px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; left: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render indeterminate progress and match snapshot 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; left: 0%; width: 100%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__indeterminate">
      <div class="v-progress-linear__indeterminate long bg-primary">
      </div>
      <div class="v-progress-linear__indeterminate short bg-primary">
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render indeterminate progress with query prop and match snapshot 1`] = `
<div class="v-progress-linear v-progress-linear--query v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; left: 0%; width: 100%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__indeterminate">
      <div class="v-progress-linear__indeterminate long bg-primary">
      </div>
      <div class="v-progress-linear__indeterminate short bg-primary">
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render reverse component in RTL mode 1`] = `
<div class="v-progress-linear v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; left: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;

exports[`VProgressLinear should render reversed component 1`] = `
<div class="v-progress-linear v-progress-linear--reverse v-theme--light"
     style="height: 4px;"
     role="progressbar"
     aria-valuemin="0"
     aria-valuemax="100"
     aria-valuenow="33"
>
  <div class="v-progress-linear__background bg-primary"
       style="opacity: 0.3; right: 33%; width: 67%;"
  >
  </div>
  <transition-stub>
    <div class="v-progress-linear__determinate bg-primary"
         style="width: 33%;"
    >
    </div>
  </transition-stub>
</div>
`;
